frontend/src/app/authentication/registration-dialog/registration-dialog.component.ts

```diff
@@ -0,0 +1,55 @@
+import {Component, OnInit} from '@angular/core';
+import {AuthenticationService} from "../../api/services";
+import {FormBuilder, FormControl, FormGroup, Validators} from "@angular/forms";
+
+@Component({
+    selector: 'app-registration-dialog',
+    templateUrl: './registration-dialog.component.html',
+    styleUrls: ['./registration-dialog.component.sass']
+})
+export class RegistrationDialogComponent implements OnInit {
+
+    public form: FormGroup;
+
+    constructor(public service: AuthenticationService, protected readonly formBuilder: FormBuilder) {
+        this.form = this.formBuilder.group({
+            name: new FormControl('', Validators.required),
+            email: new FormControl('', [Validators.required, Validators.email]),
```

Comment:
IIRC this E-Mail validator disallows non-ASCII characters like `ü,ß,é` which could potentially be part of the username component. We might want to consider relaxing the requirements to something along the lines of 

```regex
(.*)@(.*)
```